pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/DownloadParticipantListRoute.java

```diff
@@ -39,21 +39,24 @@ public class DownloadParticipantListRoute extends RequestHandler {
      * be renamed to processRequest, and the old 'processRequest' method should be deleted
      */
     public Object processRequest(Request request, Response response, String userId) throws Exception {
-        if (UserUtil.checkUserAccess(null, userId, "pt_list_export", null)) {
-            response.status(500);
-            return new Result(500, UserErrorMessages.NO_RIGHTS);
-        }
         DownloadParticipantListPayload payload =
                 ObjectMapperSingleton.instance().readValue(request.body(), DownloadParticipantListPayload.class);
         DownloadParticipantListParams params = new DownloadParticipantListParams(request.queryMap());
 
         String realm = RoutePath.getRealm(request);
+
+        String userIdReq = UserUtil.getUserId(request);
+        if (!UserUtil.checkUserAccess(realm, userId, "pt_list_view", userIdReq)) {
+            response.status(500);
+            return new Result(500, UserErrorMessages.NO_RIGHTS);
```

Comment:
slowly we are trying to remove technique of returning object of `Result`, just returning setting `response.status` and return `UserErrorMessages.NO_RIGHTS` is pretty enough

Reply:
gotcha, thanks for the heads up